include/llbuild/Core/BuildEngine.h

```diff
@@ -257,7 +257,9 @@ class BuildEngine {
   /// A database should only be attached immediately after creating the engine,
   /// it is an error to attach a database after adding rules or initiating any
   /// builds, or to attempt to attach multiple databases.
-  void attachDB(std::unique_ptr<BuildDB> database);
+  ///
+  /// \returns false if the build database could not be attached.
+  bool attachDB(std::unique_ptr<BuildDB> database);
```

Comment:
Let's report an error string here, too, so we can at least surface some minimal indication of the problem.